indexclient/client.py

```diff
@@ -320,6 +320,65 @@ def list_versions(self, did):
             versions.append(Document(self, version["did"], version))
         return versions
 
+    def query_urls_metadata(self, url, key, value, fields=None, versioned=False, limit=100, offset=0, page_size=100):
+        params = {
+            "url": url,
+            "key": key,
+            "value": value,
+            "fields": fields,
+            "versioned": versioned,
+            "limit": limit if limit < page_size else page_size,
+            "offset": offset
+        }
+
+        while limit > 0:
+            response = self._get("_query/urls/metadata/q", params=params).json()
+            if not response:
+                break
+            for entry in response:
+                yield entry
+
+            limit -= len(response)
```

Comment:
Do we need to update the params' `limit` key too? If the limit is 6 and the page_size is 5, will this return 6 or 10 Documents? We should confirm the behavior with the corresponding tests.

Reply:
Oh yeah, i think you are right